### ISSUE TITLE
Welcome plug-in shouldn't open welcome page each page load

### ISSUE THREAD

**Issue by mmorhun** [eclipse/che-theia#95]
When `@eclipse-che/welcome-plugin` is included it opens welcome page every page load. It happens each page reload even if plenty of editors are already opened. Also focus goes to that new tab, so user have to manually close it. The same problem when a user switches to dashboard and goes back to Che Theia.

**Comment by sunix** [comment]
Should open welcome page only once/the first time you run the workspace

**Comment by vitaliy-guliy** [comment]
Or just when there is no opened files

**Comment by benoitf** [comment]
as each person seems to have its own view, I don't know what should be the `expected behavior`

But we could have an option to turn it off (like VSCode)

**Comment by vitaliy-guliy** [comment]
Let's start from having this option

**Comment by benoitf** [comment]
turn it off per workspace or per user ? or user has the two choices ?

**Comment by vitaliy-guliy** [comment]
I prefer per workspace option

**Comment by vitaliy-guliy** [comment]
Disable by pressing the button or a check box on the Welcome Page

**Comment by AndrienkoAleksandr** [comment]
+1 I prefer per workspace option

**Comment by makandre** [comment]
For products that want to build on top of Che Theia and maybe supply their own welcome page, can this option be set somewhere (in the stack definition perhaps)?

**Comment by AndrienkoAleksandr** [comment]
Hello @makandre . This changes introduced to welcome-plugin. In case if you build own IDE on top of che-theia you could apply own list predefined plugins. If you don't need welcome plugin - you could postpone this plugin, or apply yours analog. che-theia has folder in the image and in this folder stored predefined plugins. CHE Theia process knows where is this folder located with help environment variables: 
`THEIA_PLUGINS`
f.e: THEIA_PLUGINS=local-dir:///plugins
So, as you can see,  plugins are not compiled in the theia binaries, they stored in the separated folder or even more - in separated container(in case remote che-theia plugin).

**Comment by makandre** [comment]
@AndrienkoAleksandr Thanks for your reply.  But could you explain what you meant by "you could postpone this plugin" or how to apply my own list of plugins?  Sorry I'm new to Che Theia

**Comment by AndrienkoAleksandr** [comment]
@makandre we have che-theia image: https://github.com/eclipse/che-theia/tree/master/dockerfiles/theia/
For ci build image, we used file where is defined list of Theia extensions and plugins https://github.com/eclipse/che-theia/blob/master/extensions/extensions.yml#L22 .
It used in logic here: https://github.com/eclipse/che-theia/blob/master/dockerfiles/theia/Dockerfile#L68 
As you can see we have che:theia tools and this tool applies plugins and images to the theia from yaml file.

**Comment by sunix** [comment]
Welcome page is provided through a Theia plugin (a .theia file), you can include it or exclude in your product or replace it by your own. You can have a look in how it is done: https://github.com/eclipse/che-theia/tree/master/plugins/welcome-plugin

**Comment by makandre** [comment]
Thanks for the info!  Just to make sure I understand, the suggested methods involve re-building/re-packaging Che Theia right?  Are there any ways to use Che Theia as-is with some runtime configuration change to include/exclude other plugins? Thanks.

**Comment by benoitf** [comment]
@makandre you could set THEIA_DEFAULT_PLUGINS to an empty variable or override it with another location (like mounting a volume where you've all your *.theia files) or mount all your customized theia files to `/default-theia-plugins` location
